classes/Product.php

```diff
@@ -3198,7 +3198,7 @@ public static function getPricesDrop(
         }
 
         if ($count) {
-            return Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
+            return (int) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
```

Comment:
This can also return false because of getValue
Check if value is not false then cast it

Reply:
Done